src/main/java/de/taimos/gpsd4java/backend/SocketThread.java

```diff
@@ -114,15 +113,14 @@ protected void retry() {
 	/**
 	 * Halts the socket thread.
 	 *
-	 * @throws InterruptedException
 	 */
-	public void halt() throws InterruptedException {
+	public void halt() {
 		this.running.set(false);
+		
 		try {
 			this.reader.close();
 		} catch (final IOException e) {
 			// ignore
 		}
-		this.join(1000);
```

Comment:
This changes makes the halt() method asynch instead of waiting for the Thread to stop. Is this intended?